scipy/stats/_sensitivity_analysis.py

```diff
@@ -0,0 +1,662 @@
+            message = (
+                f"{method!r} is not a valid 'method'. It must be one of"
+                f" {set(indices_methods)!r} or a callable."
+            )
+            raise ValueError(message) from exc
+    else:
+        kind = inspect.Parameter.POSITIONAL_OR_KEYWORD
+        parameters = (
+            inspect.Parameter('f_A', kind=kind),
+            inspect.Parameter('f_B', kind=kind),
+            inspect.Parameter('f_AB', kind=kind)
+        )
+        sig = inspect.Signature(parameters=parameters)
+        indices_method = method
+
+        if inspect.signature(indices_method) != sig:
+            message = (
+                "If 'method' is a callable, it must have the following"
+                f" signature: {inspect.signature(saltelli_2010)}"
+            )
+            raise ValueError(message)
+
+    if callable(func):
+        if dists is None:
+            raise ValueError(
+                "'dists' must be defined when 'func' is a callable."
+            )
+
+        A, B = sample_A_B(n=n, dists=dists, random_state=random_state)
+        AB = sample_AB(A=A, B=B)
+
+        f_A = np.atleast_2d(func(A))
+
+        if f_A.shape[1] != n:
+            raise ValueError(
+                "'func' output should have a shape ``(s, -1)`` with ``s`` "
+                "the number of output."
+            )
+
+        def funcAB(AB):
+            d, d, n = AB.shape
+            AB = np.moveaxis(AB, 0, -1).reshape(d, n*d)
+            f_AB = func(AB)
+            return np.moveaxis(f_AB.reshape((-1, n, d)), -1, 0)
+
+        f_B, f_AB = np.atleast_2d(func(B), funcAB(AB))
+    else:
+        message = (
+            "When 'func' is a dictionary, it must contain the following "
+            "keys: 'f_A', 'f_B' and 'f_AB'."
+            "'f_A' and 'f_B' should have a shape ``(s, n)`` and 'f_AB' "
+            "should have a shape ``(d, s, n)``."
+        )
+        try:
+            f_A, f_B, f_AB = np.atleast_2d(
+                func['f_A'], func['f_B'], func['f_AB']
+            )
+        except KeyError as exc:
+            raise ValueError(message) from exc
+
+        if f_A.shape[1] != n or f_A.shape != f_B.shape or \
+                f_AB.shape == f_A.shape or f_AB.shape[-1] % n != 0:
+            raise ValueError(message)
+
+    # Normalization by mean
+    # Sobol', I. and Levitan, Y. L. (1999). On the use of variance reducing
+    # multipliers in monte carlo computations of a global sensitivity index.
+    # Computer Physics Communications, 117(1) :52-61.
+    mean = np.mean([f_A, f_B], axis=(0, -1)).reshape(-1, 1)
+    f_A -= mean
+    f_B -= mean
+    f_AB -= mean
+
+    # Compute indices
+    # Filter warnings for constant output as var = 0
+    with np.errstate(divide='ignore', invalid='ignore'):
+        first_order, total_order = indices_method(f_A=f_A, f_B=f_B, f_AB=f_AB)
+
+    # null variance means null indices
+    first_order[~np.isfinite(first_order)] = 0
+    total_order[~np.isfinite(total_order)] = 0
```

Comment:
@ConnectedSystems this is the check for constant variance. I found it was simpler and faster to just look at non finite (nan and inf) values and replace these with 0. This is also conveniently working with vector output.

Reply:
Nice! Thank you!

Reply:
Couldn't this mask a true NaN or infinite variance?

Reply:
Discussed offline. I added a note to say function evaluations must be finite values.